tests/prompt/test_gitstatus.py

```diff
@@ -79,19 +79,14 @@ def test_gitstatus_clean(prompts, fake_proc):
 def test_no_git(prompts, fake_process, tmp_path):
     os.chdir(tmp_path)
     err = b"fatal: not a git repository (or any of the parent directories): .git"
-    for cmd in (
-        "git status --porcelain --branch",
-        "git rev-parse --git-dir",
-        "git diff --numstat",
-    ):
-        fake_process.register_subprocess(
-            command=cmd,
-            stderr=err,
-            returncode=128,
-        )
-
-    exp = ""
-    assert prompts.pick_val("gitstatus.repo_path") == ""
-    assert format(prompts.pick("gitstatus")) == exp
-    assert _format_value(prompts.pick("gitstatus"), None, None) == exp
-    assert _format_value(prompts.pick("gitstatus"), "{}", None) == exp
+    fake_process.register_subprocess(
+        command="git rev-parse --git-dir", stderr=err, returncode=128
+    )
+
+    # test that all gitstatus fields (gitstatus, gitstatus.branch,
+    # gitstatus.porceclain, etc) are None and are formatted correctly in a
+    # format string like {gitstatus: hello {}}
+    for field in prompts.get_fields(gitstatus):
+        assert prompts.pick_val(field) is None
+        assert _format_value(prompts.pick(field), None, None) == ""
+        assert _format_value(prompts.pick(field), "hello {}", None) == ""
```

Comment:
could you add a whole field case as well? 
as of now it will return empty string, since MultiPromptFIeld joins all fragments. 

xonsh/prompt/base.py:515
```
self.value = self.separator.join(self._collect(ctx))
```
This might need some update.

Reply:
I think `prompts.get_fields(gitstatus)` includes the whole field case. When I run the following in a xonsh shell
```
from xonsh.prompt import gitstatus
$PROMPT_FIELDS.get_fields(gitstatus)
list($PROMPT_FIELDS.get_fields(gitstatus)
)
```
I get the output
```
[<Prompt: gitstatus.repo_path>,
 <Prompt: gitstatus.short_head>,
 <Prompt: gitstatus.tag>,
 <Prompt: gitstatus.tag_or_hash>,
 <Prompt: gitstatus.stash_count>,
 <Prompt: gitstatus.operations>,
 <Prompt: gitstatus.porcelain>,
 <Prompt: gitstatus.branch>,
 <Prompt: gitstatus.ahead>,
 <Prompt: gitstatus.behind>,
 <Prompt: gitstatus.untracked>,
 <Prompt: gitstatus.changed>,
 <Prompt: gitstatus.deleted>,
 <Prompt: gitstatus.conflicts>,
 <Prompt: gitstatus.staged>,
 <Prompt: gitstatus.numstat>,
 <Prompt: gitstatus.lines_added>,
 <Prompt: gitstatus.lines_removed>,
 <Prompt: gitstatus.clean>,
 <Prompt: gitstatus>]
```
*(note the last line: `<Prompt: gitstatus>`)*

> as of now it will return empty string, since MultiPromptField joins all fragments.

I addressed the whole field case by overriding the `update()` method of `GitStatus`:
https://github.com/xonsh/xonsh/pull/4920/files#diff-48cdec73879d69293f5c0b4710e2ec4524d53d99ab7b511cc2492c62ea471daaR341-R345

Reply:
Good then